libraries/lib-wave-track/CachingPlayableSequence.cpp

```diff
@@ -47,11 +47,12 @@ float CachingPlayableSequence::GetChannelGain(int channel) const
 
 bool CachingPlayableSequence::Get(
    size_t iChannel, size_t nBuffers, samplePtr buffers[], sampleFormat format,
-   sampleCount start, size_t len, fillFormat fill, bool mayThrow,
-   sampleCount* pNumWithinClips) const
+   sampleCount start, size_t len, bool backwards, fillFormat fill,
+   bool mayThrow, sampleCount* pNumWithinClips) const
 {
    assert(iChannel + nBuffers <= mWaveTrack.NChannels());
-   mCacheHolders = mWaveTrack.GetSampleView(iChannel, nBuffers, start, len);
+   mCacheHolders =
+      mWaveTrack.GetSampleView(iChannel, nBuffers, start, len, backwards);
```

Comment:
I see `backwards` is used to construct the cache, but is ignored when using it.  Samples are not taken from the end and reversed.

So this implementation of `CachingPlayableSequence` is incomplete, and so in the first commit, backwards scrubbing won't work right.

But I see that later commits stop using this class, and in the remaining uses for spectrogram, there is never any backwards fetch.

So for now just `assert(!backwards)` and comment about it.

Reply:
I'm actually recommending an L principle violation with that assertion!  But I understand the fix as not worth the effort now

Reply:
I agree. It'd be hard to test, too. Spot on, though.